nixos/modules/services/desktops/pipewire/pipewire-media-session.nix

```diff
@@ -15,13 +15,15 @@ let
   defaults = {
     alsa-monitor = (builtins.fromJSON (builtins.readFile ./alsa-monitor.conf.json));
     bluez-monitor = (builtins.fromJSON (builtins.readFile ./bluez-monitor.conf.json));
+    bluez-hardware = (builtins.fromJSON (builtins.readFile ./bluez-hardware.conf.json));
     media-session = (builtins.fromJSON (builtins.readFile ./media-session.conf.json));
     v4l2-monitor = (builtins.fromJSON (builtins.readFile ./v4l2-monitor.conf.json));
   };
 
   configs = {
     alsa-monitor = recursiveUpdate defaults.alsa-monitor cfg.config.alsa-monitor;
     bluez-monitor = recursiveUpdate defaults.bluez-monitor cfg.config.bluez-monitor;
+    bluez-hardware = recursiveUpdate defaults.bluez-hardware cfg.config.bluez-hardware;
```

Comment:
Since the bulk of this is lists, which `recursiveUpdate` does not merge, it would be good to have the defaults exposed somewhow so they can be appended to customized lists.

Reply:
I feel like having some of these just be typed module options with defaults would work better? Then mkDefault, mkForce, etc would just work as expected to append/replace entries.

Reply:
I don't think that merges lists either?

Reply:
well afaik typically a list default is set as `config.whatever = [ defaults or whatever ]` (optionally with mkAfter/mkBefore) and not a default/mkOptionDefault nor mkDefault, so that assigning to the list by default appends/merges. Then mkOrder/mkForce can be used to replace the entire list if wanted.

Reply:
If it does that, great. I am admittedly not really as well acquainted with the finer points of nixos module options as I probably should.

Reply:
My experience has been that `environment.systemPackages` works this way, but  `services.pipewire.media-session.config.bluez-monitor.rules` does not. 

So, if you want to add a rule, keeping the defaults instead of replacing them, you have to look up the defaults and explicitly add them to your config (and keep them updated). (And yeah, that's probably not desired.)